waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -279,9 +280,55 @@
                 (fn []
                   (let [{:keys [active-instances failed-instances]} (:instances (service-settings waiter-url service-id))
                         pod-ids (->> (concat active-instances failed-instances)
-                                  (map :id)
-                                  (map (fn [instance-id]
-                                         (subs instance-id 0 (string/last-index-of instance-id "-"))))
-                                  (into #{}))]
+                                     (map :k8s/pod-name)
+                                     (into #{}))]
                     (log/info pod-ids)
                     (< 1 (count pod-ids)))))))))))
+
+(deftest ^:parallel ^:integration-slow ^:resource-heavy test-kubernetes-pod-expiry-grace-period
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (if-let [custom-image (System/getenv "INTEGRATION_TEST_BAD_IMAGE")]
+        (let [{:keys [container-running-grace-secs]} (get-kubernetes-scheduler-settings waiter-url)
+              request-timeout-ms (* 3 container-running-grace-secs 1000)
+              waiter-params (assoc (kitchen-params)
+                              :distribution-scheme "simple"
+                              :image custom-image
+                              :name (rand-name)
+                              :timeout request-timeout-ms)
+              waiter-headers (pc/map-keys #(str "x-waiter-" (name %)) waiter-params)
+              service-id (retrieve-service-id waiter-url waiter-headers)
+              abort-ch (async/promise-chan)
+              status-promise (promise)]
+          (is service-id)
+          (if (> container-running-grace-secs 150)
+            (is false "container-running-grace-secs will cause the test to run for too long")
+            (with-service-cleanup
+              service-id
+              ;; make request to launch service
+              (async/thread
```

Comment:
Instead of making this async thread, could we make a synchronous request instead, and have it time out after `container-running-grace-secs`? That would let us get rid of all this async channel code, and we wouldn't need the sleep either.

Reply:
I don't follow why having a connection that times out after `container-running-grace-secs` doesn't work, but sleeping for `container-running-grace-secs` does work. I assumed that any extra time needed for the scaling etc. would be covered by the `waiter-for` timeout.

Reply:
I realized that the autoscaler will try to keep at least `min-instances` running. Relying on that behavior, simplified the test even more.